File: solutions/fortran/grains/1/grains.f90

module grains

  implicit none

contains

  double precision function square(n)
    integer :: n
    if (n < 1 .or. n > 64) then
      square = -1.d0
    else
      square = 2.0**(n-1)
    end if
  end function

  double precision function total()
    integer :: n
    total=0
    do n=1,64
      total=total+square(n)
    end do
  end function

end module

File: solutions/c/grains/1/grains.c
#include "grains.h"

uint64_t square(uint8_t index){
    if (index == 1){
        return 1;
    }
    else {
        return (2*square(index-1));
    }
}

uint64_t total(){
    int i=0;
    uint64_t tot=0;
    for(i=0;i<64;i++){
        if (i==0){
            tot=square(i+1);
        }
        else{
            tot=tot+square(i+1);
        }
    }
    return tot;
}
